[FIX] stock: add translation in dynamic button contain change

mafo-odoo opened the pull request:
Steps to reproduce the issue:
1)Inventory > Configuration > Settings > Activate "Reception Report"
> Activate "Show reception report"
2)Choose a non english language
3)Sales > Choose SO > Delivery > Allocation
4)Click on the Assign/Unassign equivalent in the new language

Current behavior:
The word in the Assign/Unassign button is in english

Expected behavior:
The word in the Assign/Unassign button is in the new language

Explanation:
When clicked the button starts a function in js that changes the
button text context without translating the new text. We also modify
the text in javascript and the problem is solved.

Note for the RD stock team:
I don't think this is a good practice to modify the button content of a report dynamically.
I did a fix with the fewest changes possible but I would advise fusing the two buttons into one and 
adding a condition in it for the content, it should be possible in xml/qweb.

https://github.com/odoo/odoo/blob/d3912aad67e5a519df3b355e962ce9d2e7894133/addons/stock/report/report_stock_reception.xml#L116-L135


Issue 2:
The button had fixed size specific to the english language, if a longer
translation in another language was used it was cut leading to make it
unuderstanble in some languages (e.g. german).

opw-2827247

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/89734).

mafo-odoo commented:
@nle-odoo I changed that and I also fixed another small issue with the button width that cut the translation in some languages. Are you also okay with that?

mafo-odoo commented:
@ticodoo just for your information as you coded the concerned parts :)

nle-odoo commented:
Yes, the fixed with was done with intent in fadc2dddf12521b0f81096b6225d67951584b714 but with the button being translatable I don't think we can keep it. I don't know how necessary it was.

mafo-odoo commented:
@ticodoo what do you think about the fix size button issue with translation?

nle-odoo reviewed (approved):
seems good from tic too

robodoo r+
